### PR TITLE
MATLAB: add logging initialization function

### PR DESCRIPTION
**Pull request by sbesson**
Fixes https://github.com/openmicroscopy/bioformats/issues/2233

This PR addresses the issues reported in https://github.com/openmicroscopy/bioformats/issues/2233 by:
- adding a MATLAB function initializing the logging system with a logic similar to [logging.basicConfig](https://docs.python.org/3/library/logging.html#logging.basicConfig).
- set the default logging level to `WARN` if no argument is passed
- use this logging initialization function in `bfopen` to prevent logging set by the user to be overriden
- add some logic checking whether logging is enabled to `loci.common.DebugTools`

To test this PR:
- try the following workflow from a clear MATLAB session using the test artifacts:

 ```
% Open an image - logging should be initialized at WARN level
data = bfopen(id);
% Check whether logging is enabled
loci.common.DebugTools.isEnabled()
% Set the logging level to debug
loci.common.DebugTools.enableLogging('DEBUG');
% Open the image again - logging should be at DEBUG level
data = bfopen(id);
```

- check the new MATLAB tests are passing
- check `bftest.m` completes as expected in the [CI job](https://ci.openmicroscopy.org/view/DEV/job/BIOFORMATS-DEV-merge-matlab/) and that the individual workers in the `parfor` loop have their logging system properly intialized.

### CODE REVIEW

**Comment by bramalingam** [comment]
Works as expected, 

```
>> c1 = bfopen();
log4j:WARN No appenders could be found for logger (loci.formats.FormatHandler).
log4j:WARN Please initialize the log4j system properly.
Reading series #1
    ...
>> loci.common.DebugTools.isEnabled()

ans =

     1

>> loci.common.DebugTools.enableLogging('DEBUG');
>> c1 = bfopen();
TiffDelegateReader initializing /Users/bramalingam/Desktop/myfile.tiff
TiffDelegateReader initializing /Users/bramalingam/Desktop/myfile.tiff
loci.formats.in.TiffDelegateReader.initFile(/Users/bramalingam/Desktop/myfile.tiff)
TiffReader initializing /Users/bramalingam/Desktop/myfile.tiff
loci.formats.in.TiffReader.initFile(/Users/bramalingam/Desktop/myfile.tiff)
Reading IFDs
Populating metadata
Checking comment style
Populating OME metadata
Expected positive value for PhysicalSize; got Infinity
Reading series #1
    .Reading tile Length 8064 Offset 16
Reading tile Length 8064 Offset 8080
Reading tile Length 8064 Offset 16144
Reading tile Length 8064 Offset 24208
Reading tile Length 8064 Offset 32272
Reading tile Length 8064 Offset 40336
Reading tile Length 8064 Offset 48400
Reading tile Length 8064 Offset 56464
Reading tile Length 8064 Offset 64528
Reading tile Length 8064 Offset 72592
Reading tile Length 8064 Offset 80656
Reading tile Length 8064 Offset 88720
Reading tile Length 8064 Offset 96784
Reading tile Length 8064 Offset 104848
Reading tile Length 8064 Offset 112912
Reading tile Length 8064 Offset 120976
Reading tile Length 8064 Offset 129040
Reading tile Length 8064 Offset 137104
Reading tile Length 8064 Offset 145168
Reading tile Length 8064 Offset 153232
Reading tile Length 8064 Offset 161296
Reading tile Length 8064 Offset 169360
Reading tile Length 8064 Offset 177424
Reading tile Length 8064 Offset 185488
Reading tile Length 8064 Offset 193552
Reading tile Length 8064 Offset 201616
Reading tile Length 8064 Offset 209680
Reading tile Length 8064 Offset 217744
Reading tile Length 8064 Offset 225808
Reading tile Length 8064 Offset 233872
Reading tile Length 8064 Offset 241936
Reading tile Length 2304 Offset 250000
..
Location.mapFile: embedded-stream.raw -> null
Location.mapFile: embedded-stream.raw -> null
```

Good to merge, if the code changes suggested by @jburel  are taken care of..

**Comment by sbesson** [comment]
@bramalingam/@jburel: pushed.

File: components/formats-common/src/loci/common/DebugTools.java [hunk 1]
```diff
@@ -65,12 +65,36 @@ public static String getStackTrace(Throwable t) {
   }
 
   /**
+   * Checks wether SLF4J logging has been enabled via logback or log4j
```
Comment (mtbc) [inline components/formats-common/src/loci/common/DebugTools.java]: whether

File: components/formats-gpl/matlab/bfInitLogging.m [hunk 2]
```diff
@@ -0,0 +1,49 @@
+function bfInitLogging(varargin)
+% BFINITLOGGING initializes the Bio-Formats logging system
+%
+%   bfInitLogging() intializes the logging system at WARN level by default.
+%   It allows
```
Comment (mtbc) [inline components/formats-gpl/matlab/bfInitLogging.m]: ... ?

**Comment by jburel** [comment]
a typo missed by @mtbc ;-): ``intializes``

**Comment by sbesson** [comment]
Fixed the typo reported in d76bc0c. Assuming there are no more functional comment and as all the jobs are green, I propose to merge this PR to unlock this follow-up logging work addressing http://trac.openmicroscopy.org/ome/ticket/13151.

**Pull request merged**